llvm/lib/CodeGen/ExpandFp.cpp

```diff
@@ -948,12 +926,22 @@ static void scalarize(Instruction *I, SmallVectorImpl<Instruction *> &Replace) {
   Value *Result = PoisonValue::get(VTy);
   for (unsigned Idx = 0; Idx < NumElements; ++Idx) {
     Value *Ext = Builder.CreateExtractElement(I->getOperand(0), Idx);
-    Value *Cast = Builder.CreateCast(cast<CastInst>(I)->getOpcode(), Ext,
-                                     I->getType()->getScalarType());
-    Result = Builder.CreateInsertElement(Result, Cast, Idx);
-    if (isa<Instruction>(Cast))
-      Replace.push_back(cast<Instruction>(Cast));
+    auto *BinOp = dyn_cast<BinaryOperator>(I);
+    Value *NewOp;
```

Comment:
```suggestion
    Value *NewOp = nullptr;
```
I realise this is  assigned a value immediately but it's safer to initialise to account for future code changes.

Reply:
Right. I also changed the structure of the if-then-else to better reflect that the scalarization logic does only support BinaryOperator and Cast instructions.